dipy/workflows/align.py

```diff
@@ -45,16 +54,492 @@ def run(self, input_files, new_vox_size, order=1, mode='constant', cval=0,
             Name of the resliced dataset to be saved
             (default 'resliced.nii.gz')
         """
-        
+
         io_it = self.get_io_iterator()
 
         for inputfile, outpfile in io_it:
-            
             data, affine, vox_sz = load_nifti(inputfile, return_voxsize=True)
             logging.info('Processing {0}'.format(inputfile))
             new_data, new_affine = reslice(data, affine, vox_sz, new_vox_size,
                                            order, mode=mode, cval=cval,
                                            num_processes=num_processes)
             save_nifti(outpfile, new_data, new_affine)
             logging.info('Resliced file save in {0}'.format(outpfile))
-        
\ No newline at end of file
+
+
+class ImageRegistrationFlow(Workflow):
+    """
+    The registration workflow is organized as a collection of different
+    functions. The user can intend to use only one type of registration
+    (such as center of mass or rigid body registration only).
+
+    Alternatively, a registration can be done in a progressive manner.
+    For example, using affine registration with progressive set to 'True'
+    will involve center of mass, translation, rigid body and full affine
+    registration. Whereas, when progressive is False the registration will
+    include only center of mass and affine registration. The progressive
+    registration will be slower but will improve the quality.
+
+    This can be controlled by using the progressive flag (True by default).
+    """
+
+    def perform_transformation(self, static, static_grid2world, moving,
+                               moving_grid2world,
+                               affreg, params0, transform, starting_affine):
+
+        """ Function for translation based registration.
```

Comment:
Copy pasting is often a trap in programming.

Reply:
Yeah, totally agree. Sorry for this. Updated it. :)